qCC/ccCommandLineCommands.cpp

```diff
@@ -165,6 +165,50 @@ constexpr char OPTION_FILE_NAMES[]						= "FILE";
 constexpr char OPTION_ORIENT[]							= "ORIENT";
 constexpr char OPTION_MODEL[]							= "MODEL";
 
+bool get_SFIndex_SFName(ccCommandLineInterface& cmd, int& sfIndex, QString& sfName)
+{
+	sfIndex = -1;
+	sfName = "";
+	bool validSFIndexOrName = true;
+	sfName = cmd.arguments().takeFirst();
+	if (sfName.toUpper() == OPTION_LAST)
+	{
+		sfIndex = -2;
+		cmd.print(QObject::tr("Set active SF index: LAST"));
+	}
+	else
+	{
+		bool validInt;
+		sfIndex = sfName.toInt(&validInt);
+		if (!validInt)
+		{
+			sfIndex = -1;
+			if (sfName.size() >= 2 && sfName.startsWith('\''))
+			{
+				while (!sfName.endsWith('\'') && !cmd.arguments().empty())
+				{
+					sfName.append(' ');
+					sfName.append(cmd.arguments().takeFirst());
+				}
+				if (!sfName.endsWith('\''))
+				{
+					cmd.error(QObject::tr("Invalid SF name! (missing closing simple quote)"));
+					validSFIndexOrName = false;
+				}
+				else
+				{
+					sfName = sfName.mid(1, sfName.size() - 2);
+					cmd.print(QObject::tr("Set active SF name: '%1'").arg(sfName));
+					validSFIndexOrName = true;
+				}
+			}
+		}
+		cmd.print(QObject::tr("Set active SF index: %1").arg(sfIndex));
```

Comment:
Maybe change the message depending on whether it's an index or a name?

Reply:
ok